routers/repo/issue_dependency.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"fmt"
+	"net/http"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+)
+
+// AddDependency adds new dependencies
+func AddDependency(ctx *context.Context) {
+	depID := ctx.QueryInt64("newDependency")
+
+	issueIndex := ctx.ParamsInt64("index")
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, issueIndex)
+	if err != nil {
+		ctx.ServerError("GetIssueByIndex", err)
+		return
+	}
+
+	// Redirect
+	defer ctx.Redirect(fmt.Sprintf("%s/issues/%d", ctx.Repo.RepoLink, issueIndex), http.StatusSeeOther)
```

Comment:
Does `ctx.Flash` work with `ctx.Redirect`?
How `defer ctx.Redirect` works with `ctx.ServerError`?

Reply:
From what I know, `ctx.Flash` sets a cookie with the message to make it work with redirects.

> How defer ctx.Redirect works with ctx.ServerError?

You mean, what happens if `ctx.ServerError` throws? I think it should display the server error and omit the redirect.